### DISCUSSION TITLE
sBTC / Nakamoto: Contract Interfaces for Allowing the Node to Maintain sBTC

### DISCUSSION THREAD

**Discussion by jcnelson** [stacks-archive/sbtc-developer-release#358] [Ideas]
The Stacks node will call into sBTC to carry out various maintenance duties.  For example (this is incomplete):

* Processing consolidations to mint sBTC
* Processing withdraw transactions on BTC to burn sBTC
* Processing and clearing deposit challenges 
* Validating and clearing proof-of-withdraw BTC transactions posted by Stackers (so PoX can keep going even if the tx does not get mined)
* Activating/deactivating PoX payouts based on sBTC liveness

I think there are others

**Comment by AshtonStephens** [comment]
Needs list of things a stacks node needs to assist sBTC. We need a strict list of requirements.

Will likely determine next week in calls with @saralab @kantai @jcnelson, @soju-drinker, @jferrant etc.

**A number of these questions will be addressed in contract interactions design document that we'll post here.**

**Comment by AshtonStephens** [comment]
Need to isolate the end to end flows of the sBTC operations. Then we can sign off on the design document.
